### ISSUE TITLE
Silent message drop since the last maintenance

### ISSUE THREAD

**Issue by airone506** [matrix-org/libera-chat#6]
There is silent message drop since the **Libera.chat bridge** maintenance & upgrade on May 10th, 2023. Estimated ratio of message dropped is ~20% (based on observations, I might add exact numbers later). The Libera.chat messages are not displayed to the Matrix users in the room.

I did some checks in the past, some of them very diligent. I never observed such a massive drop. Actually, I did observe 100% message delivery even on the very diligent checks.

**Comment by airone506** [comment]
I just did quick ad-hoc "human-powered" scan of #python at Libera.Chat, going backwards in the timeline.

I saw one message in #python dropped at 06:01:29 UTC (June 8th). Text of the message was "scipy.signal", if that helps for searching.

Another message #python dropped at 04:45:34 UTC (June 8th). The text of the message was "(or this has been my experience when I've grappled with the same questions, anyway)". I did not continue with the checking after this one.

No message drop seen after that one at 06:01:29 UTC and I can tell the situation has dramatically improved indeed.

@Half-Shot: Are these two drops before some fix has been deployed?

**Comment by inglor** [comment]
> I believe this situation has now improved, but please keep letting me know if it's gotten worse.

Lost messages again this morning 08 Jun 2023 11:22:29 (according to IRC client) GMT+1 in #archlinux-aurweb at Libera.Chat. The next message appearing is 20 seconds later in both IRC and Matrix - screenshots attached.

IRC log:
![irc-log](https://github.com/matrix-org/libera-chat/assets/301857/20cb45df-f9d5-438c-b76a-6ab88cd6410c)

Matrix log:
![matrix-log](https://github.com/matrix-org/libera-chat/assets/301857/06019d7a-f986-4e4c-b9d9-1ecb166f5242)


This is not a very active channel to have multiple messages coming through.

**Comment by airone506** [comment]
I saw multiple messages being dropped in multiple rooms. Just letting to know that the issue didn't go anywhere.

**Comment by srett** [comment]
Still dropping messages as of today

**Comment by airone506** [comment]
Still dropping messages as of today.

**Comment by airone506** [comment]
Still dropping messages as of today. I'd believe lot of users who rely on the bridge as their daily driver are still unaware of this issue.

**Comment by srett** [comment]
Indeed, I start to feel this state is worse than having no bridge at all. Even if you know about this, it still leads to confusion and misunderstandings if just the right message gets dropped.

**Comment by progval** [comment]
`$1LVFwOd4_H-bpmQGdRsWLMo-t7gLHOTrXw6Wfw2k-C4` in `#irc:matrix.org`, sent today at 19:00 UTC, was not sent to Libera.

**Comment by srett** [comment]
If fixing this is too hard, maybe make the bridge just forward each message to Matrix twice. This should lower dropped messages from 10% to 1%.

**Comment by progval** [comment]
the two messages tulir posted today on `#matrix-dev:matrix.org`: `$cFi1JlMOJaUjPjEiO_NSw0PLRgAiXgcHejQnqh7A1NU` and `$HxK2VTDpIS9iiS8BcEFSBpYL3dQrGTVlBjtM8ymCU_8`, because his puppet disconnected yesterday at 16:02:59Z (that's probably the same issue as #12)

**Comment by BrenBarn** [comment]
The issue still seems to be happening.  Is there any progress?  All I see in recent messages on this ticket is people saying "the problem still exists". . .

**Comment by srett** [comment]
@BrenBarn libera staff made an announcement a month ago that they'd take some action as of July 1st, bit nothing seems to have happened. Which would mean they implicitly picked option 1: https://libera.chat/news/matrix-irc-bridge-updates

**Comment by BrenBarn** [comment]
@srett: I saw that, but as far as I can see that doesn't really have anything to do with whether this bug is being fixed.  That's just libera trying to decide how to work around the problem.

**Comment by progval** [comment]
This is still happening at the same frequency as last month. Is it useful if we keep reporting them?

**Comment by ara4n** [comment]
We haven't been updating this issue as much as we could or should; apologies. We've been dealing with the various complications around deportalling while trying to fix the  bug, and comms with the libera team has taken priority over comms with the community. Good news(?) is that we have a plausible root cause for the bug and will try to ship a fix on Monday.

I've written up a bit more context on https://news.ycombinator.com/item?id=36923504. Apologies that things got so unreliable for so long.
